compiler/resolution/visibleFunctions.cpp

```diff
@@ -643,6 +655,12 @@ static void getVisibleFunctions(const char*           name,
     if (block != rootModule->block) {
       BlockStmt* next  = getVisibilityScope(block);
 
+      ModuleSymbol* blockMod = block->getModule();
+      ModuleSymbol* nextMod = next->getModule();
+      if (blockMod != nextMod && nextMod != theProgram && nextMod != rootModule) {
+        next = standardModule->block;
+      }
```

Comment:
Should this update/check be a helper function, to avoid code drift?

Reply:
Probably.  The existing code already had issues of code duplication in these areas, which made it easy to be lazy, particularly while under a deadline.

Reply:
Fair, I'm okay with not addressing it right now, then

Reply:
I will try to do it as a follow-on PR.  With another look, it appears as though the new logic could almost be put directly into `getVisibilityScope()` although there's one use on line 301 whose purpose isn't clear to me (i.e., would it also want to skip over enclosing module scopes, or does it want the current behavior?).  Looks like code you might be familiar with, do you know?

https://github.com/chapel-lang/chapel/blob/9703984768df079dab4a69463e73cc5265077645/compiler/resolution/visibleFunctions.cpp#L298-L302

Reply:
(or maybe more to the point: What is code I could write that would demonstrate whether this call did care or not?)

Reply:
That code is specifically looking for methods and field accessors.  It ignores the privacy and limitations of use and import statements, since it would be difficult/impossible to re-enable us to see methods/field accessors left out as a result

Reply:
So probably defining a secondary method in a module accessible because of the use/import would demonstrate what you're looking for